src/AuditManager.php

```diff
@@ -44,6 +44,9 @@ public function getConfiguration()
         return $this->config;
     }
 
+    /**
+     * NEXT_MAJOR: Use `\Doctrine\ORM\EntityManagerInterface` for argument 1.
+     */
```

Comment:
```suggestion
    /**
     * NEXT_MAJOR: Use `\Doctrine\ORM\EntityManagerInterface` for argument 1.
     *
     * @return AuditReader
     */
```

I think with this there is no need of asserting array in tests (that usually happens because there is some missing return type declaration).

Reply:
The cases of the return types that weren't being respected, are those like this: https://github.com/sonata-project/EntityAuditBundle/blob/bf499c7a3ad12c1065159a384d1f53f5ae1e5133/src/AuditReader.php#L517.

Reply:
yep, but before that call there is one to `$this->auditManager->createAuditReader`, if there is no return type it's interpreted as `mixed`, then `mixed->findX` will return `mixed` as well. I'll try later, but I guess that's the problem.

Reply:
Yes. That's exactly the case. I've removed the assertions and added the return type docblock.
Thank you.